src/odb/src/db/dbSWire.cpp

```diff
@@ -164,6 +164,35 @@ void _dbSWire::addSBox(_dbSBox* box)
     callback->inDbSWireAddSBox((dbSBox*) box);
 }
 
+void _dbSWire::removeSBox(_dbSBox* box)
+{
+  _dbBlock* block = (_dbBlock*) getOwner();
+  uint boxid = box->getOID();
+  if (boxid == _wires) {
+    // at head of list, need to move head
+    _wires = (uint) box->_next_box;
+  } else {
+    // in the middle of the list, need to iterate and relink
```

Comment:
Is this a performance issue in your use model?

Reply:
I'm sure there is some, but nothing too noticeable, a doubly-linked list would solve that but since this is not something that happens often (except for this one place) it doesn't seem worth adding another field.